### ISSUE TITLE
Socket::IFF_* are missing

### ISSUE THREAD

**Issue by enebo** [jruby/jruby#8046]
Not sure how actionable these constant values are but we are missing: :IFF_ALLMULTI, :IFF_AUTOMEDIA, :IFF_BROADCAST, :IFF_DEBUG, :IFF_DYNAMIC, :IFF_LOOPBACK, :IFF_MASTER, :IFF_MULTICAST, :IFF_NOARP, :IFF_NOTRAILERS, :IFF_POINTOPOINT, :IFF_PORTSEL, :IFF_PROMISC, :IFF_RUNNING, :IFF_SLAVE, :IFF_UP.

Whether we can use these values like IFF_UP using Java APIs might be why they don't exist but perhaps we just do not have them generated?  This is currently an issue for https://github.com/tenderlove/zeroconf.

**Comment by ahorek** [comment]
is it enough to import https://github.com/jnr/jnr-constants/pull/77/files ? or do we need more?

**Comment by enebo** [comment]
@ahorek I think for the reported issue it is literally IFF_UP but I am unsure if this actually will work with what we have in Socket impl or not.  Your PR looks good to me though.  We can always add more later.

**Comment by enebo** [comment]
@ahorek Sleeping on this I wonder if we should see what other things are in Socket and consider at least generating the values for it.  Whether we can use them now or not is probably less important than never having to regenerate all these values again.   Not sure if that is a good idea or not though?

**Comment by ahorek** [comment]
here's my attempt to add missing bits https://github.com/jruby/jruby/pull/8047

but tests are still failing on
```
Java::JavaLang::NullPointerException: Cannot invoke "jnr.unixsocket.UnixSocketAddress.path()" because "unix" is null
    org.jruby.dist/org.jruby.ext.socket.RubyBasicSocket.local_address(RubyBasicSocket.java:596)
```

https://github.com/jruby/jruby/blob/627d8d1667fbf111f0c3c31b5eb8f35cef96ca16/core/src/main/java/org/jruby/ext/socket/RubyBasicSocket.java#L596

this is because our setsockopt implementation is limited and the expected IP isn't set, see
https://github.com/tenderlove/zeroconf/blob/c87d39571da4d4a54d0e989995e47ab23e4aea46/lib/zeroconf/utils.rb#L57
and
https://github.com/jruby/jruby/blob/627d8d1667fbf111f0c3c31b5eb8f35cef96ca16/core/src/main/java/org/jruby/ext/socket/RubyBasicSocket.java#L517

ruby: #<UDPSocket:fd 5, AF_INET, 172.26.115.118, 41991>
jruby: #<UDPSocket:fd 12>

so there's still some work to do

**Comment by ahorek** [comment]
> Whether we can use them now or not

we do have values for Socket::IFF_* but don't define them in JRuby. In case there's an optional feature like TCP cork, we can skip it by faking missing support
```
if Socket.const_defined?(:TCP_CORK)
      socket.setsockopt()...
```
and the program will still work. However, it won't make a difference if the program depends on the feature like in this case.

**Comment by enebo** [comment]
@ahorek I misread the original link thinking that PR hadn't been merged yet so I figured this would be something where we were doing regeneration on platforms.  I think we should wait on other flags since this is already done.

I landed your other PR to make IFF_* and use IP_* values.

More work to be done but we are closer.  Perhaps we should open a new issue on what you have encountered and close this out since you added the values for IFF.  This sounds like it may be potentially more than one issue since one is supporting some IFF features and the other is working towards UDP multicast for zeroconf.
